graders/c/cgrader/cgrader.py

```diff
@@ -476,20 +476,28 @@ def compile_re(t):
         elif msg is None:
             msg = ""
 
-        points = True
+        points = max_points
         if timeout and "TIMEOUT" in outcmp:
-            points = False
+            points = 0
         elif size_limit and len(outcmp) > size_limit:
             out = out[0:size_limit] + "\nTRUNCATED: Output too long."
-            points = False
+            points = 0
+        elif any(r.search(outcmp) is not None for _, r in reject_output):
+            points = 0
+        elif must_match_all_outputs == "partial":
+            points = (
+                max_points
+                * len([_ for _, r in exp_output if r.search(outcmp) is not None])
```

Comment:
```suggestion
                * sum(1 if r.search(outcmp) is not None else 0 for _, r in exp_output)
```
Slight nitpick but I think makes the code clearer.